constraints/src/main/java/gov/nasa/jpl/aerie/constraints/tree/Expression.java

```diff
@@ -2,16 +2,20 @@
 
 import gov.nasa.jpl.aerie.constraints.model.ActivityInstance;
 import gov.nasa.jpl.aerie.constraints.model.SimulationResults;
+import gov.nasa.jpl.aerie.constraints.time.Window;
 
 import java.util.Map;
 import java.util.Set;
 
 public interface Expression<T> {
-  T evaluate(final SimulationResults results, final Map<String, ActivityInstance> environment);
+  T evaluate(final SimulationResults results, final Window bounds, final Map<String, ActivityInstance> environment);
```

Comment:
🔶 I do think this is a good approach - one alternative that comes to mind is to have the caller set the `SimulationResults.bounds` field before calling `evaluate`, but I think your way is better.

Tagging @kroffo and @Twisol in case they have any reactions to this.